creativecommons/wp-plugin-creativecommons
# Add wp-cli in docker image

## Fixes
- Fixes #200  by @possumbilities

## Description

Adds wp-cli support in docker image for developement and testing.

## Technical details

In order to install wp-cli we needed a custom Docker image. So this pull request creates a docker image in Dockerfile. 

An alternate approach would be to push an image directly to docker hub. But there is no benefit of adding the complication of maintaining the image on docker hub.
## Tests

1. Checkout the PR.
2. Remove the data volumes:
run the following commands in `wp-content/plugins/wp-plugin-creativecommons/dev/`
    ```shell
    sudo rm -rf db/.* db/* wordpress/.* wordpress/*
    docker-compose up -d
    ```
3. The output of ` docker-compose up -d ` should look like following:
    ```
    Creating network "dev_default" with the default driver
    Building wordpress
    [+] Building 1.9s (8/8) FINISHED                                                                                                                              docker:default
     => [internal] load build definition from Dockerfile                                                                                                                    0.0s
     => => transferring dockerfile: 536B                                                                                                                                    0.0s
     => [internal] load metadata for docker.io/library/wordpress:latest                                                                                                     0.0s
     => [internal] load .dockerignore                                                                                                                                       0.0s
     => => transferring context: 2B                                                                                                                                         0.0s
     => CACHED [1/4] FROM docker.io/library/wordpress:latest                                                                                                                0.0s
     => [2/4] RUN curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar &&     chmod +x wp-cli.phar &&     mv wp-cli.phar /usr/local/bin/wp     1.2s
     => [3/4] WORKDIR /var/www/html                                                                                                                                         0.1s 
     => [4/4] RUN wp --info                                                                                                                                                 0.3s 
     => exporting to image                                                                                                                                                  0.1s 
     => => exporting layers                                                                                                                                                 0.1s
     => => writing image sha256:171d767e88b15b4b9e23b51af3ff6984d03a9b7b9e89c64eabf1284e056a06c4                                                                            0.0s 
     => => naming to docker.io/library/dev_wordpress                                                                                                                        0.0s 
    WARNING: Image for service wordpress was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.   
    Creating local-wordpress-db ... done
    Creating local-wordpress    ... done
    ```

## Screenshots

![image](https://github.com/creativecommons/wp-plugin-creativecommons/assets/74826173/0b8e3aee-ce4f-4052-a5cf-64795d35632e)

## Checklist



  index.md`).
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin


For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
